### ISSUE TITLE
Online robust PCA methods

### ISSUE THREAD

**Issue by tjof2** [hyperspy/hyperspy#1457]
ORPCA currently takes 4(!) method options:

```
method : {None, 'CF', 'BCD', 'SGD', 'MomentumSGD'}
        'CF'  - Closed-form solver
        'BCD' - Block-coordinate descent
        'SGD' - Stochastic gradient descent
        'MomentumSGD' - Stochastic gradient descent with momentum
```

`SGD` and `MomentumSGD` are my own work to deal with changing subspaces (e.g. sample drift or contrast reversal when denoising a video sequence), while `CF` and `BCD` are transferred over from the code written in Matlab by the original authors, and don't cope well with the issues mentioned above. Howevver, both are very good options for denoising e.g. EELS/EDS datasets where the subspace doesn't change, and in fact typically perform better than `SGD` in those cases.

There are plans afoot to publish the `SGD`/`MomentumSGD` methods, but there is also another method I have code for that avoids the usual machine learning issue with stochastic gradient descent: *that of selecting an appropriate step size/learning rate*. 

Currently the user has little control over the step size apart from selecting an initial parameter - it then uses a hard-coded exponential function to reduce the step size over time, which again is fine for EELS/EDS but not good for a video where there might be an unpredictable change (e.g. scene shift, moving around the sample, etc.). 

The alternative method avoids this issue entirely, but I'm conscious it would then result in **5 methods** for ORPCA. Of course clear documentation would help, but ultimately the "correct" method to use depends on the particular application. Thoughts welcome...

**Comment by thomasaarholt** [comment]
I think clear documentation is absolutely necessary (I've spent far too many hours trying to decipher what on earth the difference between the various python linear fitting algorithms is!), but a **"quick-start" list of recommended settings for a few different applications** would be most useful, and take the fear of programming out of potential users.

**Comment by tjof2** [comment]
Sure, I get that - one of the advantages of the new method is that it removes a difficult parameter to select so makes it more user-friendly.

The other question of course is what the best "default" should be...

**Comment by thomasaarholt** [comment]
I would probably suggest the user friendly method being the default. Then, if users want to be more specialised, then they can try the other algorithms.

**Comment by thomasaarholt** [comment]
@tjof2 Any progress on this?

**Comment by tjof2** [comment]
When I get internet in my new flat...gimme until the end of the universe maybe?

**Comment by thomasaarholt** [comment]
You're telling me you don't have a 4G chip implanted in your skull? :D
